src/libraries/System.Runtime/tests/System/Attributes.cs

```diff
@@ -295,10 +295,21 @@ public static void GetCustomAttributesWorksWithOpenAndClosedGenericTypesForEvent
             GenericAttributesTestHelper<DateTime?>(t => Attribute.GetCustomAttributes(@event, t));
         }
 
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/56887", TestRuntimes.Mono)]
+        public static void GetCustomAttributesOnOpenGenericTypeRetrievesDerivedAttributes()
+        {
+            Attribute[] attributes = Attribute.GetCustomAttributes(typeof(HasGenericAttribute), typeof(GenericAttribute<>));
+            Assert.Equal(2, attributes.Length);
+            Assert.Equal(1, attributes.OfType<DerivesFromGenericAttribute>().Count());
+            Assert.Equal(1, attributes.OfType<GenericAttribute<string>>().Count());
+        }
+
```

Comment:
NIT: Could we have a few more tests like for other member types also some scenarios like having same attribute with different type parameters and requesting an attribute with a concrete type would return only that requested one (of course if that scenario is not covered)

Reply:
> Could we have a few more tests like for other member types

We have tests for all the member types calling into `GenericAttributesTestHelper` which tests open generic attributes.

> same attribute with different type parameters and requesting an attribute with a concrete type would return only that requested one

Added more explicit coverage for this case.

Reply:
Great, thanks!